runtime/doc/lua.txt

```diff
@@ -1740,6 +1818,26 @@ list_slice({list}, {start}, {finish})                       *vim.list_slice()*
     Return: ~
         (list) Copy of table sliced from start to finish (inclusive)
 
+map({f}, {src}, {...})                                             *vim.map()*
```

Comment:
> @gpanders any objection to changing this signature so that the function is the last param? That is generally the pattern we should strive for, because the function is usually the biggest part of a call (lexically).
> 
> ```
> map({src}, {f})
> ```
> 
> Though if the variadic `...` arg is needed then this can't change. Is the variadic arg part of `{src}` ?

@mfussenegger and I [discussed this as well](https://github.com/neovim/neovim/pull/23029#discussion_r1167067295). I also prefer trailing function for stylistic reasons, but as you pointed out we need the variadic params in order for this to work properly (a Lua function iterator uses multiple values to maintain its state. For example, the `pairs()` function actually returns 3 values, `:h luaref-in`).